src/plugin/index-jexl-filter/src/test/org/apache/nutch/indexer/filter/TestJexlIndexingFilter.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nutch.indexer.filter;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.apache.nutch.crawl.CrawlDatum;
+import org.apache.nutch.crawl.Inlinks;
+import org.apache.nutch.indexer.NutchDocument;
+import org.apache.nutch.metadata.Metadata;
+import org.apache.nutch.parse.Outlink;
+import org.apache.nutch.parse.ParseData;
+import org.apache.nutch.parse.ParseImpl;
+import org.apache.nutch.parse.ParseStatus;
+import org.apache.nutch.util.NutchConfiguration;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestJexlIndexingFilter {
+
+	@Test
+	public void testBasicIndexingFilter() throws Exception {
+		Configuration conf = NutchConfiguration.create();
+		conf.set("index.jexl.filter", "doc.lang[0]=='en'");
+
+		JexlIndexingFilter filter = new JexlIndexingFilter();
+		filter.setConf(conf);
+		Assert.assertNotNull(filter);
+
+		NutchDocument doc = new NutchDocument();
+
+		String title = "The Foo Page";
+		Outlink[] outlinks = new Outlink[] { new Outlink("http://foo.com/", "Foo") };
+		Metadata metaData = new Metadata();
+		metaData.add("Language", "en/us");
+		ParseData parseData = new ParseData(ParseStatus.STATUS_SUCCESS, title, outlinks, metaData);
+		ParseImpl parse = new ParseImpl("this is a sample foo bar page. hope you enjoy it.", parseData);
+
+		CrawlDatum crawlDatum = new CrawlDatum();
+		crawlDatum.setFetchTime(100L);
+
+		Inlinks inlinks = new Inlinks();
+
+		doc.add("lang", "en");
+
+		try {
+			NutchDocument result = filter.filter(doc, parse, new Text("http://nutch.apache.org/index.html"), crawlDatum, inlinks);
+			Assert.assertNotNull(result);
+			Assert.assertEquals(doc, result);
+		} catch (Exception e) {
+			e.printStackTrace();
+			Assert.fail(e.getMessage());
+		}
+		
+		doc.removeField("lang");
+		doc.add("lang", "ru");
+		
+		try {
+			NutchDocument result = filter.filter(doc, parse, new Text("http://nutch.apache.org/index.html"), crawlDatum, inlinks);
+			Assert.assertNull(result);
+		} catch (Exception e) {
+			e.printStackTrace();
+			Assert.fail(e.getMessage());
```

Comment:
When this check is moved to a different test method you should use the exception assertion from JUnit, instead of the `try ... catch` block.

Reply:
I'm not sure I understand your comment. If I understand correctly, I should just remove the try...catch, and let the Exception be thrown, since that will fail the test with the exception.
For the new (third) test, where we want to create an exception, I assume we should use ExpectedException.
Agreed?
(BTW, all these problems with the test were also in the test you pointed me to as a template!)

Reply:
@pipldev Yes for testing the exception you could use something like

```java
@Test(expected = IndexOutOfBoundsException.class) 
public void empty() { 
     new ArrayList<Object>().get(0); 
}
```
The idea is that if you isolate this into a separated test you don't need the try ... catch because for that particular test you'll know if it's supposed to throw an exception or not (hence my suggestion of having separated tests). So for test `A` you know it's supposed to work as expected and you don't need to handle the exception because you know that when you're setting the data for the filter. And on test `B` you know that it's supposed to throw an exception because you made sure of that when configuring the filter and your assertions will reflect that. 

We try to follow the _boy scout rule_ when some code is touched/added we try to leave it in a better form than how we found it :). With this approach, we try to keep the stability of Nutch and make the life easier for the maintainers. And yes, some pieces of our code could use some love :)